airbyte-webapp/src/components/connectorBuilder/Builder/PaginationSection.tsx

```diff
@@ -0,0 +1,180 @@
+        options={injectIntoValues}
+        label="Inject into"
+        tooltip="Configures where the page size should be set on the HTTP requests"
+      />
+      <BuilderField
+        type="string"
+        path={streamFieldPath("paginator.pageSizeOption.field_name")}
+        label="Field name"
+        tooltip="Configures which key should be used in the location that the page size is being injected into"
+        optional
+      />
+    </GroupControls>
+  );
+
+  return (
+    <BuilderCard
+      toggleConfig={{
+        label: (
+          <ControlLabels
+            label="Pagination"
+            infoTooltipContent="Configure how pagination is handled by your connector"
+          />
+        ),
+        toggledOn,
+        onToggle: handleToggle,
+      }}
+    >
+      <BuilderOneOf
+        path={streamFieldPath("paginator.strategy")}
+        label="Mode"
+        tooltip="Pagination method to use for requests sent to the API"
+        options={[
+          {
+            label: "Offset Increment",
+            typeValue: "OffsetIncrement",
+            children: (
+              <>
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Size of pages to request from API"
+                />
+                {pageTokenOption}
+                {pageSizeOption}
+              </>
+            ),
+          },
+          {
+            label: "Page Increment",
+            typeValue: "PageIncrement",
+            children: (
+              <>
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Size of pages to request from API"
+                />
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.start_from_page")}
+                  label="Start from page"
+                  tooltip="Page number to start requesting pages from"
+                  optional
+                />
+                {pageTokenOption}
+                {pageSizeOption}
+              </>
+            ),
+          },
+          {
+            label: "Cursor Pagination",
+            typeValue: "CursorPagination",
+            children: (
+              <>
+                <BuilderField
+                  type="string"
+                  path={streamFieldPath("paginator.strategy.cursor_value")}
+                  label="Cursor value"
+                  tooltip="Value of the cursor to send in requests to the API"
+                />
+                <BuilderField
+                  type="number"
+                  path={streamFieldPath("paginator.strategy.page_size")}
+                  label="Page size"
+                  tooltip="Size of pages to request from API"
+                  optional
```

Comment:
It seems like we have a problem with optional number fields - clearing them out will turn them into an empty string which will fail the validation.